### ISSUE TITLE
Recent update throwing an error `X-Frame-Options to SAMEORIGIN`

### ISSUE THREAD

**Issue by chrisnicola** [ryanb/letter_opener#105]
Just recently updated Rails to 4.2.1 and I'm seeing this. Not sure what might be causing it. Is this a known issue? I'm guessing this is related to it using an iFrame. I'm not sure what changed though.

```
Refused to display 'http://localhost:3000/confirm-email?email_confirmation_token=dBLTjydYe_XgqEdDRufseQ' in a frame because it set 'X-Frame-Options' to 'SAMEORIGIN'.
```

**Comment by nashby** [comment]
Can you please provide a sample application that reproduces the error?

**Comment by seanlinsley** [comment]
The default Rails application throws this error. Specifically because Rails 4(.1?) sets `X-Frame-Options` on every request to `SAMEORIGIN`

**Comment by chrisnicola** [comment]
Sorry I have not had a chance to get back to this. I did read somewhere, that adding `target="_blank"` to any urls might be one solution.

**Comment by nashby** [comment]
@seanlinsley I just tried it with a new Rails 4.2 app and still can't reproduce it. :(

Is it any chance you can provide steps for reproducing it? Thanks!

**Comment by jtokoph** [comment]
##### Steps to create

1. Create a new rails app with letter_opener configured
2. Send a mailer with a link to a page within the app (eg http://localhost:3000).
3. Click the link in the mailer window
4. Request goes through, but browser does not display the content. Browser still displays email content.

##### Issue

letter_opener displays email content in an iframe within an html file served via file:// protocol.
Clicking a link in an email displayed via letter_opener will try to load the link within the iframe.
The browser will see that the x-frame-options header is set to SAMEORIGIN and compares the null origin of file:// with the website origin of localhost and blocks display of the frame.

##### Potential solutions

- Display email headers in an iframe and email content at top level
  - This will sandbox the headers from the email css, but could mess with email display
- Rewrite all links in letter_opener views to open in new window or at the top level
  - Parsing and replacing isn't necessarily trivial
- Prepend `<base target="_top">`  to email content before display

**Comment by nashby** [comment]
@jtokoph thank you very much for such great comment! I've pushed a fix (`<base>` tag solution). Can you guys please test your application with letter_opener from the master branch? Thanks!

**Comment by jtokoph** [comment]
@nashby Just confirmed that the master branch works. :shipit: Thanks for the quick turnaround!

**Comment by chrisnicola** [comment]
Thanks @jtokoph and @nashby that's awesome. Really appreciate this, sorry I wasn't much help. I'll test this asap.

**Comment by nashby** [comment]
Hey guys, I've just released 1.4.0 version with this fix. Please test it and report any issues back! Thanks.

**Comment by chrisnicola** [comment]
This does seem to fix the problem. I also noticed that letter opener no longer has a max-width? Was this an intentional style change?
